tests/integration_tests/test_types.py

```diff
@@ -0,0 +1,329 @@
+    assert res, err
+
+# NOT IMPLEMENTED
+# def test_getTransactionRaw(ethermint, geth):
+#     eth_rpc = ethermint.w3.provider
+#     geth_rpc = geth.w3.provider
+#     eth_rsp, geth_rsp = make_same_rpc_calls(eth_rpc, geth_rpc, "eth_getRawTransactionByHash", ["0x5c504ed432cb51138bcf09aa5e8a410dd4a1e204ef84bfed1be16dfba1b22060"])
+#     res, err = same_types(eth_rsp, geth_rsp)
+#     print(eth_rsp)
+#     print(geth_rsp)
+#     assert res, err
+
+def test_getTransactionReceipt(ethermint, geth):
+    eth_rpc = ethermint.w3.provider
+    geth_rpc = geth.w3.provider
+    make_same_rpc_calls(eth_rpc, geth_rpc, "eth_getTransactionReceipt", ["0x5c504ed432cb51138bcf09aa5e8a410dd4a1e204ef84bfed1be16dfba1b22060"])
+
+    # FIXME: 'type': '0x0' not in expected, status key on json is not in ethermint
+    tx_hash = send_and_get_hash(ethermint.w3)
+
+    tx_res = eth_rpc.make_request('eth_getTransactionReceipt', [tx_hash])
+    expected = {
+    'blockHash': '0x4e3a3754410177e6937ef1f84bba68ea139e8d1a2258c5f85db9f1cd715a1bdd',
+    'blockNumber': 46147,
+    'contractAddress': None,
+    'cumulativeGasUsed': 21000,
+    'from': '0xA1E4380A3B1f749673E270229993eE55F35663b4',
+    'gasUsed': 21000,
+    'logs': [],
+    'logsBloom': '0x000000000000000000000000000000000000000000000000...0000',
+    'status': 1, # 0 or 1
+    'to': '0x5DF9B87991262F6BA471F09758CDE1c0FC1De734',
+    'transactionHash': '0x5c504ed432cb51138bcf09aa5e8a410dd4a1e204ef84bfed1be16dfba1b22060',
+    'transactionIndex': 0,
+    }
+    res, err = same_types(tx_res['result'], expected)
+    assert res, err
+
+def test_feeHistory(ethermint, geth):
+    eth_rpc = ethermint.w3.provider
+    geth_rpc = geth.w3.provider
+    make_same_rpc_calls(eth_rpc, geth_rpc, "eth_feeHistory", [4, 'latest',[10,90]])
+
+    make_same_rpc_calls(eth_rpc, geth_rpc, "eth_feeHistory", [4, '0x5000',[10,90]])
+
+    fee_history = eth_rpc.make_request('eth_feeHistory', [4, 'latest',[10,90]])
+    expected = {
+        'oldestBlock': 3,
+        'reward': [[220, 7145389], [1000000, 6000213], [550, 550], [125, 12345678]],
+        'baseFeePerGas': [202583058, 177634473, 155594425, 136217133, 119442408],
+        'gasUsedRatio': [0.007390479689642084, 0.0036988514889990873, 0.0018512333048507866, 0.00741217041320997]
+    }
+    res, err = same_types(fee_history['result'], expected)
+    assert res, err
+
+def test_estimateGas(ethermint, geth):
+    tx = {"to": ADDRS["community"], "from": ADDRS["validator"]}
+
+    eth_rpc = ethermint.w3.provider
+    geth_rpc = geth.w3.provider
+    make_same_rpc_calls(eth_rpc, geth_rpc, "eth_estimateGas", [tx])
+    make_same_rpc_calls(eth_rpc, geth_rpc, "eth_estimateGas", [{}])
+
+def make_same_rpc_calls(rpc1, rpc2, method, params):
+    res1 = rpc1.make_request(
+        method, params
+    )
+    res2 = rpc2.make_request(
+        method, params
+    )
+    res, err = same_types(res1, res2)
+    assert res, err
+
+def same_types(objectA, objectB):
```

Comment:
Does it only compare types, not values?

Reply:
yes, since we are not really concerned about error messages beeing different and geth doesnt have the same chain behind it.

we can check values later if we want to.